x-pack/osquerybeat/beater/osquerybeat.go

```diff
@@ -128,6 +129,12 @@ func (bt *osquerybeat) close() {
 
 // Run starts osquerybeat.
 func (bt *osquerybeat) Run(b *beat.Beat) error {
+	pj, err := proc.CreateJobObject()
```

Comment:
Is this a bug fix? Why was this necessary?

Reply:
this was needed in osquerybeat in order to avoid having orphaned child osqueryd process
https://learn.microsoft.com/en-us/windows/win32/procthread/job-objects
"Operations performed on a job object affect all processes associated with the job object. Examples include enforcing limits such as working set size and process priority or terminating all processes associated with a job."